consensus/istanbul/backend/snapshot.go

```diff
@@ -185,7 +190,7 @@ func (s *Snapshot) UnmarshalJSON(b []byte) error {
 	s.Epoch = j.Epoch
 	s.Number = j.Number
 	s.Hash = j.Hash
-	s.ValSet = validator.NewSet(j.Validators)
+	s.ValSet = validator.NewSetWithCache(j.Validators)
```

Comment:
here's something strange. Not related to this PR actually.

A ValidatorSet is actually a `([]validators, randomness)`, but when we "save" it we don't save the randomness, and thus on unmarshall we don't have it, nor we care.

Which makes me wonder if we really want a validatorSet or if we should save the randomenss.

Reply:
Looks like randomness is only needed in the round state to determine next proposer, so it's not needed in the snapshots.